### PR TITLE
New Kyma component list from new installation

### PR DESCRIPTION
**Pull request by jasiu001**
Changes proposed in this pull request:

- Adjust component list provider from KEB to a new artifact with Kyma components which is a part of a new installation tool

**Related issue(s)**
#600

### CODE REVIEW

**Comment by jasiu001** [comment]
TODO (before merge):
- bump KEB image
- change the source of library (control-plane/provisioner) from fork to the origin

File: components/kyma-environment-broker/internal/runtime/provider.go [hunk 1]
```diff
@@ -19,68 +19,111 @@ import (
...
 	// Read mounted config (path)
 	managedRuntimeComponents, err := r.getManagedRuntimeComponents()
 	if err != nil {
-		return nil, errors.Wrap(err, "while getting managed runtime components list")
+		return components, errors.Wrap(err, "while getting managed runtime components list")
 	}
 
-	// Return merged list, managed components added at the end
-	merged := append(openSourceKymaComponents, managedRuntimeComponents...)
+	// Add managed compinents from config to open source Kyma components
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/runtime/provider.go:90]:
```suggestion
	// Add managed components from config to open source Kyma components
```
```suggestion
	// Add managed compinents from config to open source Kyma components
```

File: components/kyma-environment-broker/internal/runtime/provider_test.go [hunk 2]
```diff
@@ -110,55 +155,79 @@ func TestRuntimeComponentProviderGetFailures(t *testing.T) {
...
 	}
 }
 
+type ListDecider struct {
+	newComponentList bool
+}
+
+func (ld *ListDecider) IsNewComponentList(version string) (bool, error) {
+	return ld.newComponentList, nil
+}
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/runtime/provider_test.go:188]: can we use the ficture you've already provided in `fixture.go`?:
```golang
type FakeListDecider struct {
	New bool
}
```

if there is a reason for redeclaring this fake logic here, you can remove unused parameter 'version' in IsNewComponentList method

**Comment by ksputo** [comment]
Please rebase with latest master as go modules migration in KEB was merged and
```
components/kyma-environment-broker/Gopkg.lock
components/kyma-environment-broker/Gopkg.toml
```
were deleted

File: components/kyma-environment-broker/internal/newinstallation/switcher.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/pkg/errors"
+	coreV1 "k8s.io/api/core/v1"
+	k8sErr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	configMapName      = "new-installation-versions"
+	configMapNamespace = "kcp-system"
```
[the remark this answers is not in the document]
Comment (jasiu001) [reply components/kyma-environment-broker/internal/newinstallation/switcher.go:16]: Do you suggest env or another place?

[the remark this answers is not in the document]
Comment (ksputo) [reply components/kyma-environment-broker/internal/newinstallation/switcher.go:16]: I'd suggest env - what I mean is a scenario when we might want to use other configMap, probably with custom KEB deployment or something. Then we won't have to rebuild the image with the changed value, just pass another CM name as env variable.

**Comment by jasiu001** [comment]
Deprecated.

**Pull request closed without merging**